source/fundamentals/promises.txt

```diff
@@ -0,0 +1,194 @@
+======================
+Promises and Callbacks
+======================
+
+.. default-domain:: mongodb
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 1
+   :class: singlecols
+
+Overview
+--------
+
+The NodeJS driver uses the asynchronous Javascript API to communicate with
+your MongoDB cluster.
+
+Asynchronous Javascript allows you to execute operations without waiting for
+the processing thread to become free. This helps prevent your application
+from becoming unresponsive to web UI rendering or network requests when
+executing long-running operations. For more information about asynchronous
+Javascript, see the MDN web documentation on
+`Asynchronous Javascript <https://developer.mozilla.org/en-US/docs/Learn/JavaScript/Asynchronous>`_.
+
+This section describes two features of asynchronous Javascript --
+`Promises`_ and `Callbacks`_ -- that you can use with the NodeJS driver to
+access the results of your method calls to your MongoDB cluster.
+
+Promises
+--------
+
+A Promise is an object returned by the asynchronous method call that allows
+you to access information on the eventual success or failure of the operation.
+They are often used to execute operations that take some time and have
+when you want to notify once it's completed. For more information on Promises
```

Comment:
```suggestion
They are used to execute operations upon the completion of other operations that take
an indeterminate time and have. For more information on Promises
```

Reply:
I think this could be misleading to the reader because:
-It's not a requirement for the operation to take an indeterminate time to use Promises
-It's not a requirement to use Promises for operations that take an indeterminate time to run
-I think there's a sentence fragment here: "and have".
-There's no mention about notification once it's completed which is a key feature of what it replaces/wraps (callbacks)